test/rules/avoid_types_as_parameter_names.dart

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_types_as_parameter_names`
+
+typedef void f1(); // OK
+typedef void f2(int a); // OK
+typedef void f3(int); // LINT
+typedef void f4(
+  num a, // OK
+  {
+  int, // LINT
+});
+typedef void f5(
+  double a, // OK
+  [
+  bool, // LINT
+]);
```

Comment:
We should have at least one test that uses the generic function type syntax:
```
typedef f6 = int Function(int); // OK
```

Reply:
Done